app/Http/Requests/AddCalendarEvent.php

```diff
@@ -31,7 +31,9 @@ public function rules()
             'year' => 'required',
             'length' => 'required|integer|min:1',
             'is_recurring' => 'nullable',
-            'recurring_until' => 'nullable',
+            'recurring_until' => 'required_with:recurring_until_month|required_with:recurring_until_day|nullable',
```

Comment:
Worth adding the `integer` rule on all 3 to avoid any shenanigans from API users.